impl_core/src/main/java/io/opencensus/implcore/metrics/MetricRegistryImpl.java

```diff
@@ -17,85 +17,71 @@
 package io.opencensus.implcore.metrics;
 
 import static com.google.common.base.Preconditions.checkNotNull;
+import static io.opencensus.implcore.internal.Utils.checkListElementNotNull;
 
 import io.opencensus.common.Clock;
-import io.opencensus.common.ToDoubleFunction;
-import io.opencensus.common.ToLongFunction;
-import io.opencensus.implcore.metrics.Gauge.DoubleGauge;
-import io.opencensus.implcore.metrics.Gauge.LongGauge;
 import io.opencensus.metrics.LabelKey;
-import io.opencensus.metrics.LabelValue;
+import io.opencensus.metrics.LongGauge;
 import io.opencensus.metrics.MetricRegistry;
 import io.opencensus.metrics.export.Metric;
 import io.opencensus.metrics.export.MetricProducer;
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Collections;
-import java.util.LinkedHashMap;
 import java.util.LinkedHashSet;
+import java.util.List;
 import java.util.Set;
+import javax.annotation.Nullable;
 
 /** Implementation of {@link MetricRegistry}. */
 public final class MetricRegistryImpl extends MetricRegistry {
   private final RegisteredMeters registeredMeters;
   private final MetricProducer metricProducer;
+  private volatile Set<String> registeredMetrics = Collections.emptySet();
 
   MetricRegistryImpl(Clock clock) {
     registeredMeters = new RegisteredMeters();
     metricProducer = new MetricProducerForRegistry(registeredMeters, clock);
   }
 
   @Override
-  public <T> void addLongGauge(
-      String name,
-      String description,
-      String unit,
-      LinkedHashMap<LabelKey, LabelValue> labels,
-      T obj,
-      ToLongFunction<T> function) {
-    checkNotNull(labels, "labels");
-    registeredMeters.registerMeter(
-        new LongGauge<T>(
-            checkNotNull(name, "name"),
+  public LongGauge addLongGauge(
+      String name, String description, String unit, List<LabelKey> labelKeys) {
+    checkNotNull(name, "name");
```

Comment:
This is inconsistent with the Noop implementation. If you do checks in this class then do the checks in the MetricRegistry as well not in the NoopLongGauge.